src/porepy/utils/comp_geom.py

```diff
@@ -473,6 +473,10 @@ def remove_edge_crossings(vertices, edges, tol=1e-3, verbose=0, snap=True,
     NotImplementedError if a 3D point array is provided.
 
     """
+    # Sanity check of input specification edge endpoints
+    assert np.all(np.diff(edges[:2], axis=0) != 0), 'Found point edge before'\
```

Comment:
We can try with this. It may result in strange errors other places, but that doesn't imply your thinking is wrong.